experimental/personnames/src/api.rs

```diff
@@ -0,0 +1,187 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use std::mem::{discriminant, Discriminant};
+
+use bitmask_enum::bitmask;
+
+use icu_locid::Locale;
+
+/// Trait for providing person name data.
+pub trait PersonName {
+    /// Return the name locale of person name.
+    fn name_locale(&self) -> Option<&Locale>;
+
+    /// return the preferred order of person name.
+    fn preferred_order(&self) -> Option<&PreferredOrder>;
+
+    /// return the value of the given field name, it *must* match the name field.
+    fn get(&self, field: &NameField) -> Option<&str>;
+
+    /// Return all available name field.
+    fn available_name_fields(&self) -> Vec<&NameField>;
+}
+
+/// Implementation for all PersonName struct.
+///
+/// see https://www.unicode.org/reports/tr35/tr35-personNames.html#person-name-object
+impl dyn PersonName {
```

Comment:
yeah, I moved some function around, but i would still like to have the validate function as not part of the API since it's defined in spec what is a valid structure or not. which sparks the question 
how to you enforce a user to build a struct that validate a pattern, when they can do anything they want ? 

e.g.: 
```
// trust me
fn is_valid() {
 true
} 
```